File: e267_ordinal_places/js/ord.js
/**
 * e267
 * With bonus
 */

// get argument, dog position then maximum (default to 100)
const position = process.argv[2];
const max = parseInt(process.argv[3]) + 1 || 101;

// construct result set
var result = [];
for (let i = 1; i < max; i++) {
    ord = "";
    if (i % 10 == 1) {
        ord = "st";
    } else if (i % 10 == 2) {
        ord = "nd";
    } else if (i % 10 == 3) {
        ord = "rd";
    } else {
        ord = "th";
    }

    //special case dealing with 11, 12 and 13
    if (i == 11 || i == 12 || i == 13) {
        ord = "th";
    }

    // exclude the position
    if (i != position) {
        result.push(i + ord);
    }
}

// Print Result
console.log( result.join(', ') );

File: e267_ordinal_places/python/ord.py
import sys

def main(position, max):
    position = int(position)

    # exclude 0th by starting at 1
    max = int(max) + 1
    result = []
    for i in range(1, max):
        ord = ""
        if i % 10 == 1:
            ord = "st"
        elif i % 10 == 2:
            ord = "nd"
        elif i % 10 == 3:
            ord = "rd"
        else:
            ord = "th"

        # special case 11, 12, 13
        if i == 11 or i == 12 or i == 13:
            ord = "th"

        # exclude position
        if i != position:
            result.append(str(i)+ord)

    print ', '.join(result)

max = 100
if (len(sys.argv) > 2):
    max = int(sys.argv[2])
main( int(sys.argv[1]), max);